src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -754,7 +754,11 @@ private void handleEditorTweaksReceived(JSONObject tweaksMessage) {
         private void handleVariantsReceived(JSONArray variants) {
             final SharedPreferences preferences = getSharedPreferences();
             final SharedPreferences.Editor editor = preferences.edit();
-            editor.putString(SHARED_PREF_CHANGES_KEY, variants.toString());
+            if(variants != null) {
```

Comment:
We could change the code in ```DecideMessages.java``` to set ```variants``` as an empty ```JSONArray``` and then check here if the array is empty. If so, then we could also set the ```SHARED_PREF_CHANGES_KEY``` to be empty. Really it's a choice between null and empty sets in either case as the end code won't change, just the conditional logic to enter the code.